source/class/qx/io/jsonrpc/Client.js

```diff
@@ -0,0 +1,323 @@
+/* ************************************************************************
+
+   qooxdoo - the new era of web development
+
+   http://qooxdoo.org
+
+   Copyright:
+      2020 Christian Boulanger
+
+   License:
+     MIT: https://opensource.org/licenses/MIT
+     See the LICENSE file in the project's top-level directory for details.
+
+   Authors:
+     * Christian Boulanger (cboulanger)
+
+************************************************************************ */
+
+/**
+ * This class provides a JSON-RPC client object with auto-configuration of the
+ * transport used (based on the URI passed).
+ */
+qx.Class.define("qx.io.jsonrpc.Client",
+{
+  extend : qx.io.transport.AbstractClient,
+
+  statics: {
+    // statics are not inherited from parent class
+    registerTransport : qx.io.transport.AbstractClient.registerTransport
```

Comment:
Thanks for the corrections and the feedback! As to the static question, it cannot be a member since it's a classic static method that manages a "global" state, not instance data. If that copy-to-subclass operation appears problematic, the solution would be to use `qx.io.transport.AbstractClient.registerTransport` directly, although I personally think the shortcut is more convenient even if it involves the copy operation.

Reply:
Correction: using `qx.io.transport.AbstractClient.registerTransport` wouldn't work, since the [registry is "global" per child class](https://github.com/cboulanger/qooxdoo/blob/merge-new-io-high-level-apis/source/class/qx/io/transport/AbstractClient.js#L43). So I'd rather leave it the way it is.

Reply:
Unless I'm missing something, it can still be a member method that manipulates the global state in AbstractClass. Why wouldn't that work?

Reply:
I still don't understand, but this is a minor issue. Ignoring.

Reply:
Ah, sorry, I meant to get back to this question but it wasn't part of the last commits. I do see your point but am glad if it's acceptable to you as it is. My repsonse would be that yes, the method could be called on an instance and still manipulate some class-wide state, but it would mean to create this instance for only this purpose, which for me kind of indicates that this is the purpose of an abstract method. The fact that statics aren't inherited in qooxdoo seems to be a qooxdoo-specific issue. It's not the case in Java, for example: https://stackoverflow.com/questions/10291949/are-static-methods-inherited-in-java

Reply:
Got it. Yeah, you're right, if registration is done without any instance in existence, it's a bit silly to instantiate an instance just to call a single method. Ship it!